lib/committee/middleware/base.rb

```diff
@@ -24,70 +21,29 @@ def call(env)
       end
     end
 
-    private
-
-    # For modern use of the library a schema should be an instance of
-    # Committee::Drivers::Schema so that we know that all the computationally
-    # difficult parsing is already done by the time we try to handle any
-    # request.
-    #
-    # However, for reasons of backwards compatibility we also allow schema
-    # input to be a string, a data hash, or a JsonSchema::Schema. In the former
-    # two cases we just parse as if we were sent hyper-schema. In the latter,
-    # we have the hyper-schema driver wrap it in a new Committee object.
-    def get_schema(options)
-      schema = options[:schema]
-      unless schema
-        schema = Committee::Drivers::load_from_file(options[:filepath]) if options[:filepath]
-
-        raise(ArgumentError, "Committee: need option `schema` or filepath") unless schema
-      end
-
-      # These are in a separately conditional ladder so that we only show the
-      # user one warning.
-      if schema.is_a?(String)
-        warn_string_deprecated
-      elsif schema.is_a?(Hash)
-        warn_hash_deprecated
-      elsif schema.is_a?(JsonSchema::Schema)
-        warn_json_schema_deprecated
-      end
+    class << self
+      def get_schema(options)
+        schema = options[:schema]
+        unless schema
+          schema = Committee::Drivers::load_from_file(options[:filepath]) if options[:filepath]
```

Comment:
As mentioned in #203, let's try to do a bulk rename from `filepath` to `schema_path`.